src/functions/public/Export-Json.ps1

```diff
@@ -0,0 +1,163 @@
+        Export-Json -InputObject $data -Path 'config.json' -IndentationType Spaces -IndentationSize 2
+
+        Exports data to config.json with 2-space indentation.
+
+        .EXAMPLE
+        Export-Json -JsonString $jsonText -Path 'data.json' -Compact
+
+        Exports a JSON string to data.json in compact format.
+
+        .EXAMPLE
+        $objects | Export-Json -Path 'output-{0}.json'
+
+        Exports multiple objects to numbered files via pipeline.
+
+        .EXAMPLE
+        Export-Json -InputObject $config -Path 'settings.json' -IndentationType Tabs -Force
+
+        Exports configuration to settings.json with tab indentation, overwriting if it exists.
+
+        .LINK
+        https://psmodule.io/Json/Functions/Export-Json/
+    #>
+
+    [CmdletBinding(DefaultParameterSetName = 'FromObject', SupportsShouldProcess)]
+    param (
+        # PowerShell object to convert and export as JSON.
+        [Parameter(Mandatory, ValueFromPipeline, ParameterSetName = 'FromObject')]
+        [PSObject]$InputObject,
+
+        # JSON string to export to file.
+        [Parameter(Mandatory, ValueFromPipeline, ParameterSetName = 'FromString')]
+        [string]$JsonString,
+
+        # The path to the output JSON file. Supports placeholders for pipeline processing.
+        [Parameter(Mandatory)]
+        [string]$Path,
+
+        # Produce compact (minified) output.
+        [Parameter()]
+        [switch]$Compact,
+
+        # Indentation type: 'Spaces' or 'Tabs'.
+        [Parameter()]
+        [ValidateSet('Spaces', 'Tabs')]
+        [string]$IndentationType = 'Spaces',
+
+        # Number of spaces or tabs per indentation level. Only used if not compacting.
+        [Parameter()]
+        [UInt16]$IndentationSize = 4,
+
+        # The maximum depth to serialize nested objects.
+        [Parameter()]
+        [int]$Depth = 2,
+
+        # Overwrite existing files without prompting.
+        [Parameter()]
+        [switch]$Force,
+
+        # Text encoding for the output file.
+        [Parameter()]
+        [ValidateSet('ASCII', 'BigEndianUnicode', 'BigEndianUTF32', 'OEM', 'Unicode', 'UTF7', 'UTF8', 'UTF8BOM', 'UTF8NoBOM', 'UTF32')]
+        [string]$Encoding = 'UTF8NoBOM'
+    )
+
+    begin {
+        $fileIndex = 0
+    }
+
+    process {
+        try {
+            # Determine the input object
+            $objectToExport = if ($PSCmdlet.ParameterSetName -eq 'FromString') {
+                $JsonString | ConvertFrom-Json -Depth $Depth -ErrorAction Stop
+            } else {
+                $InputObject
+            }
+
+            # Generate the file path (support for placeholders in pipeline scenarios)
+            $outputPath = if ($Path -match '\{0\}') {
+                $Path -f $fileIndex
+                $fileIndex++
+            } else {
+                $Path
+            }
+
+            # Resolve the full path
+            $resolvedPath = $ExecutionContext.SessionState.Path.GetUnresolvedProviderPathFromPSPath($outputPath)
```

Comment:
Can you do something more readable than this? This seems like a strange way to resolve a path. Why not just use Resolve-Path?